Meilidown/Controllers/DocumentsController.cs

```diff
@@ -48,9 +54,61 @@ public DocumentsController(IIndexingService indexingService)
             }
         }
 
+
+        var orderFiles = _sourcesProvider.GetSources()
+                                         .SelectMany(source => source.FindFiles(new("\\.order")))
+                                         .ToDictionary(sf => sf.RelativePath.Split('/')[0..^1],
+                                                       sf => sf);
+
+        await SetOrder(tree, orderFiles);
+
         return tree;
     }
 
+    private async Task SetOrder(Dictionary<string, NavNode> tree, Dictionary<string[], SourceFile> orderFiles)
+    {
+        int i = 0;
+
+        foreach (var treeNode in tree)
+            await SetOrderByParent(treeNode.Value, orderFiles, i++);
+    }
+
+    [NonAction]
+    private async Task SetOrderByParent(NavNode node, Dictionary<string[], SourceFile> orderFiles, int index)
+    {
+        node.order = index;
+
+        if (node.children != null)
+        {
+            int childIndex = 0;
+
+            // recurse all children
+            foreach (var treeNode in node.children)
+                await SetOrderByParent(treeNode.Value, orderFiles, childIndex++);
+
+            var path = node.location.Split('/');
+
+            // if this particular folder has a .order, override the order
+            var orderFile = orderFiles.SingleOrDefault(of => of.Key.SequenceEqual(path));
+
+            if (orderFile.Value != null)
+            {
+                var lines = await System.IO.File.ReadAllLinesAsync(orderFile.Value.AbsolutePath);
```

Comment:
`using System.IO.File;`

Reply:
I could do `using static System.IO.File`, but that's ugly. I can't do `using System.IO` because `File` is already a thing.

Maybe this should be moved to a `IFileOrderService`.